Error: class template placeholder 'container_over_dna4_test' not permitted in this context

Hi all.

## Platform

  - SeqAn version: 3.0.3-rc.1
  - Operating system: Fedora 35
  - Compiler: GCC-11.1.1

## Description

```
[  2%] Building CXX object CMakeFiles/seqan3_test_header_test--seqan3-test-range-container_test_template-no-self-include.dir/seqan3_test_header_test_files/seqan3/test/range/container_test_template.hpp-no-self-include.cpp.o
/usr/bin/ccache /usr/lib64/ccache/g++ -DSEQAN3_DISABLE_DEPRECATED_WARNINGS -DSEQAN3_HAS_BZIP2=1 -DSEQAN3_HAS_ZLIB=1 -DSEQAN3_HEADER_TEST -DSEQAN3_WITH_CEREAL=1 -DSEQAN3_WITH_LEMON=1 -I/usr/include/gtest -I/builddir/build/BUILD/seqan3/test/include -I/builddir/build/BUILD/seqan3/include -isystem /builddir/build/BUILD/seqan3/submodules/sdsl-lite/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -std=c++20 -DNDEBUG -pedantic -Wall -Wextra -Werror -pthread -falign-loops=32 -MD -MT CMakeFiles/seqan3_test_header_test--seqan3-test-range-container_test_template-no-self-include.dir/seqan3_test_header_test_files/seqan3/test/range/container_test_template.hpp-no-self-include.cpp.o -MF CMakeFiles/seqan3_test_header_test--seqan3-test-range-container_test_template-no-self-include.dir/seqan3_test_header_test_files/seqan3/test/range/container_test_template.hpp-no-self-include.cpp.o.d -o CMakeFiles/seqan3_test_header_test--seqan3-test-range-container_test_template-no-self-include.dir/seqan3_test_header_test_files/seqan3/test/range/container_test_template.hpp-no-self-include.cpp.o -c /builddir/build/BUILD/seqan3/build-header/seqan3_test_header_test_files/seqan3/test/range/container_test_template.hpp-no-self-include.cpp
/builddir/build/BUILD/seqan3/build-header/seqan3_test_header_test_files/seqan3/test/range/container_test_template.hpp-no-self-include.cpp:338:47: error: class template placeholder 'container_over_dna4_test' not permitted in this context
  338 | GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST(container_over_dna4_test);
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/seqan3/build-header/seqan3_test_header_test_files/seqan3/test/range/container_test_template.hpp-no-self-include.cpp:338:47: note: use 'auto' for an abbreviated function template
/builddir/build/BUILD/seqan3/build-header/seqan3_test_header_test_files/seqan3/test/range/container_test_template.hpp-no-self-include.cpp:338:72: error: expected constructor, destructor, or type conversion before ';' token
  338 | GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST(container_over_dna4_test);
      |                                                                        ^
make[2]: *** [CMakeFiles/seqan3_test_header_test--seqan3-test-range-container_test_template-no-self-include.dir/build.make:84: CMakeFiles/seqan3_test_header_test--seqan3-test-range-container_test_template-no-self-include.dir/seqan3_test_header_test_files/seqan3/test/range/container_test_template.hpp-no-self-include.cpp.o] Error 1
```

Full build log: https://kojipkgs.fedoraproject.org//work/tasks/372/68420372/build.log

### How to repeat the problem

I'm compiling seqan3 against system `gtest-1.10.0`, `range-v3-0.11.0`, `cereal-1.3.0`.
Configuration:

```
+ /usr/bin/cmake -S . -B x86_64-redhat-linux-gnu -DCMAKE_C_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_Fortran_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DLIB_SUFFIX=64 -DBUILD_SHARED_LIBS:BOOL=ON -S test/header -B build-header -DCMAKE_BUILD_TYPE=Release -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DSEQAN3_CEREAL:BOOL=ON -DSEQAN3_LEMON:BOOL=ON -DSEQAN3_TEST_BUILD_OFFLINE:BOOL=ON
-- The CXX compiler identification is GNU 11.1.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib64/ccache/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Finding SeqAn3 and checking requirements:
--   SeqAn3 include dir found:   /builddir/build/BUILD/seqan3/include
--   Detected as running from a repository checkout…
--     …adding submodule include:  /builddir/build/BUILD/seqan3/submodules/sdsl-lite/include
--   C++ Standard-17 support:    builtin
--   C++ Concepts support:       builtin
--   C++ Filesystem header:      <filesystem>
--   C++ Filesystem library:     builtin
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
--   Thread support:             via -pthread
--   Required dependency:        Range-V3 found.
--   Required dependency:        SDSL found.
--   Required dependency:        Cereal found.
--   Required dependency:        Lemon found.
--   Optional dependency:        ZLIB-1.2.11 found.
--   Optional dependency:        BZip2-1.0.8 found.
--   Optional dependency:        libexecinfo found.
--   SeqAn3 platform.hpp build:  passed.
-- Found SeqAn3: /builddir/build/BUILD/seqan3/include  
-- Finding program ccache
-- Finding program ccache - Success
-- Found GTest: /usr/lib64/cmake/GTest/GTestConfig.cmake (found version "1.10.0")  
-- Configuring done
-- Generating done
```

--- a/test/include/seqan3/test/range/container_test_template.hpp
+++ b/test/include/seqan3/test/range/container_test_template.hpp
@@ -333,7 +333,7 @@ REGISTER_TYPED_TEST_SUITE_P(container_over_dna4_test,
                             resize,
                             serialisation);
 
-#ifdef SEQAN3_HEADER_TEST
+#if defined(SEQAN3_HEADER_TEST) && defined(GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST)
 // header test will throw an unused warning otherwise
 GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST(container_over_dna4_test);
-#endif // SEQAN3_HEADER_TEST
+#endif // defined(SEQAN3_HEADER_TEST) && defined(GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST)

